### DISCUSSION TITLE
Trouble with Uptime Kuma - super slow responses

### DISCUSSION THREAD

**Discussion by dreaded369** [caprover/caprover#1951] [Q&A] [answered]
I've deployed Uptime Kuma using the one-click. I've also run it on on a local machine and a VPS running coolify, so I got a little experience.

Checking uptime of sites in the same datacentre I'm getting pings over over 1 second and occasional timeouts.
Running on my local server and coolify (same datacentre) gets around 30ms with no timeouts.

I've installed NetData and nothing looks amiss - low traffic and CPU utilisation.

Any hints on where I should look next? Is there anything about CapRover that would lead to such behaviour?

**Comment by githubsaturn** [comment]
There is nothing specific about CapRover, specially for outgoing requests (from Kuma container to another apps), even nginx isn't involved. 

Try manually shelling into the container and run curl command to see if you're seeing a long ping as well.

**Reply by dreaded369** [reply]
> Try manually shelling into the container and run curl command to see if you're seeing a long ping as well.

I just tried this:
from home - 20ms
from VPS shell - 200ms
from uptime kuma - 1200ms
So there is definitely something up with the VPS, but only enough to explain a 10X slowdown, not a 60X slowdown.

**Reply by githubsaturn** [reply]
> from uptime kuma - 1200ms

Is this from the uptime kuma shell or the uptime kuma panel? You can shell into the container itself and run curl from within the container. 
Is that what you did to get 1200?

https://caprover.com/docs/troubleshooting.html#how-to-run-shell-inside-my-application-inside-container

**Reply by dreaded369** [reply]
The original was from the uptime-kuma panel.
I just exec'd into the container and got the same result as the vps shell - 200ms

**Reply by githubsaturn** [reply]
hmm i honestly don't know what could be the reason beyond the uptime kuma code itself.... it's possible that the http client library that uptime kuma uses somehow is suboptimal in the docker swarm environment... but yea there's really nothing standing out as a point of failure in your setup

**Reply by dreaded369** [reply, marked as answer]
Here are some things I've worked out:
1. my initial 'from home' 20ms number was a ping, not a get request like I'm asking uptime kuma to do. Uptime Kuma from home using a GET request is ~200ms
2. I went back to my coolify install to double check something and it was giving the same very slow responses. Bad troubleshooting from me :(
3. I had been deploying on Vultr. I deployed on Digital Ocean and I'm getting 25ms for a ping and 200ms for a GET request. So something in the Vultr stack is killing reponse time.
4. The default CapRover install has 'enable websockets' enabled, but it's not auto-selected when you do an install. Further it needs to be enabled or uptime-kuma won't work (also the default version could be bumped)
5. There are extra settings provided by Uptime Kuma for operating behind NGINX - https://github.com/louislam/uptime-kuma/wiki/Reverse-Proxy - I haven't made the changes and everything seems fine.

So all good for me I'll just use Digital Ocean (and look at moving to something bigger but still cheap like Contabo)

**Reply by silasabbott** [reply]
@dreadedhamish If you're looking around still, I migrated to Hetzner from DigitalOcean recently and have been pretty happy with it and the lower prices

**Reply by dreaded369** [reply]
thanks - I moved to Contabo and it's way faster. Same setup:
vultr - 1200ms
digitalocean - 200ms
contabo - between 60-90ms
it could have been a noisy neighbour on vultr.
